modules/contentbox/models/system/CBHelper.cfc

```diff
@@ -760,7 +760,169 @@ component accessors="true" singleton threadSafe{
 	}
 
 
+	/**
+	* Set the Meta Canonical URL for the request
+	* @url - The new url 
+	*/
+	function setMetaURL( required string url ){
+		var prc = getPrivateRequestCollection();
+		checkMetaStruct();
+		prc.meta.url = arguments.url;
+	}
+	
+	/**
+	* Get the Meta Canonical URL for the request
+	*/
+	function getMetaURL(){
+		var prc = getPrivateRequestCollection();
+		checkMetaStruct();
+		if( structKeyExists( prc.meta, "url" ) ){
+			return prc.meta.url;
+		} else {
+			return '';
+		}
+	}
+	
+
+	/**
+	* Get the Canonical URL based on content type
+	*/
+	function getContentURL(){
+		var oCurrentContent = "";
+		var oCurrentEntryPoint = "";
+		
+		if( len( getMetaURL() ) ){
+			return getMetaURL();
+		}
+		
+		// Check if in page view or entry view
+		if( isPageView() ){
+			oCurrentContent = getCurrentPage();
+			oCurrentEntryPoint = "";
+		} else if( isEntryView() ){
+			oCurrentContent = getCurrentEntry();
+			oCurrentEntryPoint = setting( 'cb_site_blog_entrypoint' ) & "/";
+		}
+
+		// in context view or global
+		if( isObject( oCurrentContent ) AND len( oCurrentContent.getslug() ) ){
+			return siteBaseURL() & oCurrentEntryPoint & oCurrentContent.getslug();
+		}
+	}
+	
+	/**
+	* Set the Meta ImageURL for the request
+	* @ImageURL - The new ImageURL 
+	*/
+	function setMetaImageURL( required string ImageURL ){

```

Comment:
Its basically an Image URL to be used in the META information... mainly for the open graph.